x/ibc-account/keeper/handshake.go

```diff
@@ -77,5 +60,49 @@ func (k Keeper) OnChanOpenTry(
 		return sdkerrors.Wrap(channeltypes.ErrChannelCapabilityNotFound, err.Error())
 	}
 
+	// Register interchain account if it does not already exist
+	_, _ = k.RegisterInterchainAccount(ctx, portID)
+	return nil
+}
+
+func (k Keeper) OnChanOpenAck(
+	ctx sdk.Context,
+	portID,
+	channelID string,
+	counterpartyVersion string,
+) error {
+	k.SetActiveChannel(ctx, portID, channelID)
+	k.SetInterchainAccountAddress(ctx, portID)
+
+	return nil
+}
+
+// Set active channel
+func (k Keeper) OnChanOpenConfirm(
+	ctx sdk.Context,
+	portID,
+	channelID string,
+) error {
+
 	return nil
 }
+
+// May want to use these for re-opening a channel when it is closed
+//// OnChanCloseInit implements the IBCModule interface
+//func (am AppModule) OnChanCloseInit(
+//	ctx sdk.Context,
+//	portID,
+//	channelID string,
+//) error {
+//	// Disallow user-initiated channel closing for transfer channels
+//	return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "user cannot close channel")
+//}
+
+//// OnChanCloseConfirm implements the IBCModule interface
+//func (am AppModule) OnChanCloseConfirm(
+//	ctx sdk.Context,
+//	portID,
+//	channelID string,
+//) error {
+//	return nil
```

Comment:
"unset" active channel

Reply:
I'm thinking we can create some issues for the work that is left to do. I.e. unset active channels + additional checks in the channel callbacks.